File: back/src/main/java/org/example/back/db/entity/Character.java
package org.example.back.db.entity;

import java.util.LinkedHashSet;
import java.util.Set;

import org.example.back.common.BaseEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.Lob;
import jakarta.persistence.OneToMany;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "`character`")
public class Character extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name", length = 30)
    private String name;

    @Lob
    @Column(name = "detail")
    private String detail;

    @Lob
    @Column(name = "img_url")
    private String imgUrl;

	@OneToOne
	@JoinColumn(name="achievement_id")
	private Achievement achievement;

}

File: front_android/lib/src/model/character.dart
class Character {
  int id;
  String detail;
  String imgUrl;
  String name;
  int achievementId;
  bool isCheck;
  bool unlockStatus;

  Character({
    required this.id,
    required this.detail,
    required this.imgUrl,
    required this.name,
    required this.achievementId,
    required this.isCheck,
    required this.unlockStatus,
  });

  factory Character.fromJson(Map<String, dynamic> json) {
    return Character(
      detail: json['detail'],
      id: json['id'],
      imgUrl: json['imgUrl'],
      name: json['name'],
      achievementId: json['achievementId'],
      isCheck: json['isCheck'],
      unlockStatus: json['unlockStatus'],
    );
  }
}
